qiskit/qpy/binary_io/circuits.py

```diff
@@ -226,8 +226,8 @@ def _read_instruction(
             use_symengine=use_symengine,
         )
     if circuit is not None:
-        qubit_indices = dict(enumerate(circuit.qubits))
-        clbit_indices = dict(enumerate(circuit.clbits))
+        qubit_indices = circuit.qubits
+        clbit_indices = circuit.clbits
     else:
         qubit_indices = {}
         clbit_indices = {}
```

Comment:
I was going to ask if we could make the second set of `qubit_indices` and `clbit_indices` lists as well to avoid having a weird typing distinction, but actually the only play they're used is immediately below this block in the _next_ `if circuit is not None` bit.

Can we remove the `else` and fold the next `if` into this one?

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11944/commits/3395259aabf0369fa26a1e2ed8abfd24b6268dee